llvm/lib/Target/Sparc/SparcSubtarget.h

```diff
@@ -82,6 +88,13 @@ class SparcSubtarget : public SparcGenSubtargetInfo {
     return is64Bit() ? 2047 : 0;
   }
 
+  bool isRegisterReserved(MCPhysReg PhysReg) const {
+    if (PhysReg >= SP::G0 && PhysReg <= SP::O7)
+      return ReserveRegister[PhysReg - SP::G0];
```

Comment:
I wouldn't bother "compacting" a BitVector. This would also allow reserving other registers in the future and not depend on exact values of enum members.

Reply:
I think so. `MRI::getNumRegs()` used in `SparcRegisterInfo::getReservedRegs()` returns the same value.